packages/machine-extractor/src/MachineExtractResult.ts

```diff
@@ -708,8 +708,9 @@ export class MachineExtractResult {
   };
 
   modify(edits: Array<MachineEdit>) {
-    // we need to read it before calling `recast.parse` as that mutates the AST and removes comments
+    // we need to read it before calling `recast.parse` as that mutates the AST and removes comments and sometimes also locations
     const existingLayoutComment = this.getLayoutComment();
+    const existingMachineNodeLoc = this.machineCallResult.node.loc!;
```

Comment:
I don't know why this case was special - but I've confirmed that `recast` is sometimes removing `.loc` from nodes. So we have to save the existing `loc` here - before we pass the whole thing to `recast` 🙈

Reply:
Nice find 🕵🏻